controllers/machinedeploymentupgrade_controller.go

```diff
@@ -57,3 +126,95 @@ func (r *MachineDeploymentUpgradeReconciler) SetupWithManager(mgr ctrl.Manager)
 		For(&anywherev1.MachineDeploymentUpgrade{}).
 		Complete(r)
 }
+
+func (r *MachineDeploymentUpgradeReconciler) reconcile(ctx context.Context, log logr.Logger, mdUpgrade *anywherev1.MachineDeploymentUpgrade) (ctrl.Result, error) {
+	log.Info("Upgrading all worker nodes")
+	for _, machineRef := range mdUpgrade.Spec.MachinesRequireUpgrade {
+		nodeUpgrade := mdNodeUpgrader(machineRef, kubernetesVersion)
```

Comment:
can we get the node Upgrade object first to see if it exists and handle the condition accordingly instead of attempting to create the object in every reconcile ?